tests/scene/test_path_follow_2d.h

```diff
@@ -32,205 +32,222 @@
+	curve->set_bake_interval(1);
 	curve->add_point(Vector2(0, 0));
 	curve->add_point(Vector2(100, 0));
 	curve->add_point(Vector2(100, 100));
 	curve->add_point(Vector2(0, 100));
 	curve->add_point(Vector2(0, 0));
-	const Path2D *path = memnew(Path2D);
+	Path2D *path = memnew(Path2D);
 	path->set_curve(curve);
-	const PathFollow2D *path_follow_2d = memnew(PathFollow2D);
+	PathFollow2D *path_follow_2d = memnew(PathFollow2D);
+	path_follow_2d->set_loop(false);
 	path->add_child(path_follow_2d);
+	SceneTree::get_singleton()->get_root()->add_child(path);
 
 	path_follow_2d->set_progress(0);
-	CHECK(path_follow_2d->get_transform().get_origin().is_equal_approx(Vector2(0, 0)));
+	CHECK(is_equal_approx(Vector2(0, 0), path_follow_2d->get_transform().get_origin()));
 
 	path_follow_2d->set_progress(50);
-	CHECK(path_follow_2d->get_transform().get_origin().is_equal_approx(Vector2(50, 0)));
+	CHECK(is_equal_approx(Vector2(50, 0), path_follow_2d->get_transform().get_origin()));
 
 	path_follow_2d->set_progress(100);
-	CHECK(path_follow_2d->get_transform().get_origin().is_equal_approx(Vector2(100, 0)));
+	CHECK(is_equal_approx(Vector2(100, 0), path_follow_2d->get_transform().get_origin()));
 
 	path_follow_2d->set_progress(150);
-	CHECK(path_follow_2d->get_transform().get_origin().is_equal_approx(Vector2(100, 50)));
+	CHECK(is_equal_approx(Vector2(100, 50), path_follow_2d->get_transform().get_origin()));
 
 	path_follow_2d->set_progress(200);
-	CHECK(path_follow_2d->get_transform().get_origin().is_equal_approx(Vector2(100, 100)));
+	CHECK(is_equal_approx(Vector2(100, 100), path_follow_2d->get_transform().get_origin()));
 
 	path_follow_2d->set_progress(250);
-	CHECK(path_follow_2d->get_transform().get_origin().is_equal_approx(Vector2(50, 100)));
+	CHECK(is_equal_approx(Vector2(50, 100), path_follow_2d->get_transform().get_origin()));
 
 	path_follow_2d->set_progress(300);
-	CHECK(path_follow_2d->get_transform().get_origin().is_equal_approx(Vector2(0, 100)));
+	CHECK(is_equal_approx(Vector2(0, 100), path_follow_2d->get_transform().get_origin()));
 
 	path_follow_2d->set_progress(350);
-	CHECK(path_follow_2d->get_transform().get_origin().is_equal_approx(Vector2(0, 50)));
+	CHECK(is_equal_approx(Vector2(0, 50), path_follow_2d->get_transform().get_origin()));
 
 	path_follow_2d->set_progress(400);
-	CHECK(path_follow_2d->get_transform().get_origin().is_equal_approx(Vector2(0, 0)));
+	CHECK(is_equal_approx(Vector2(0, 0), path_follow_2d->get_transform().get_origin()));
 
 	memdelete(path);
 }
 
-TEST_CASE("[PathFollow2D] Removal of a point in curve") {
-	const Ref<Curve2D> &curve = memnew(Curve2D());
-	curve->add_point(Vector2(0, 0));
-	curve->add_point(Vector2(100, 0));
-	curve->add_point(Vector2(100, 100));
-	const Path2D *path = memnew(Path2D);
-	path->set_curve(curve);
-	const PathFollow2D *path_follow_2d = memnew(PathFollow2D);
-	path->add_child(path_follow_2d);
+// TEST_CASE("[SceneTree][PathFollow2D] Removal of a point in curve") {
```

Comment:
```suggestion
//TEST_CASE("[SceneTree][PathFollow2D] Removal of a point in curve") {
```
No space before commented out code, for all

Reply:
I've commented these tests because of https://github.com/godotengine/godot/pull/51372#discussion_r724466895. I don't think the behavior these tests intended to test was ever implemented in `PathFollow{2D,3D}` so we may want to remove these tests instead.

Reply:
With the following adjustments it passes:p. We need to do the same immediate update as for the other test cases, and after the curve length has changed, the absolute progress value is outdated (and doesn't represent the ratio of 0.5 anymore).
```cpp
	path_follow_3d->set_progress_ratio(0.5);
	path_follow_3d->update_transform(true);
	CHECK(is_equal_approx(Vector3(100, 0, 0), path_follow_3d->get_transform().get_origin()));

	curve->remove_point(1);
	path_follow_3d->set_progress_ratio(0.5);
	path_follow_3d->update_transform(true);
	CHECK_MESSAGE(
			is_equal_approx(Vector3(50, 50, 0), path_follow_3d->get_transform().get_origin()),
			"Path follow's position should be updated after removing a point from the curve");
```